dataframely/columns/array.py

```diff
@@ -0,0 +1,128 @@
+        primary_key: Literal[False] = False,
+        check: Callable[[pl.Expr], pl.Expr] | None = None,
+        alias: str | None = None,
+        metadata: dict[str, Any] | None = None,
+    ):
+        """
+        Args:
+            inner: The inner column type. No validation rules on the inner type are supported yet.
+            shape: The shape of the array.
+            nullable: Whether this column may contain null values.
+            primary_key: Whether this column is part of the primary key of the schema.
+                Not yet supported for the Array type.
+            check: A custom check to run for this column. Must return a non-aggregated
+                boolean expression.
+            alias: An overwrite for this column's name which allows for using a column
+                name that is not a valid Python identifier. Especially note that setting
+                this option does _not_ allow to refer to the column with two different
+                names, the specified alias is the only valid name.
+            metadata: A dictionary of metadata to attach to the column.
+        """
+        if primary_key:
+            raise ValueError(
+                "`primary_key=True` is not yet supported for the Array type."
+            )
+
+        if inner.primary_key or (
+            isinstance(inner, Struct)
+            and any(col.primary_key for col in inner.inner.values())
+        ):
+            raise ValueError(
+                "`primary_key=True` is not yet supported for inner types of the Array type."
+            )
+
+        # We disallow validation rules on the inner type since Polars arrays currently don't support .eval(). Converting
+        # to a list and calling .list.eval() is possible, however, since the shape can have multiple axes, the recursive
+        # conversion could have significant performance impact. Hence, we simply disallow inner validation rules.
+        # Another option would be to allow validation rules only for sampling, but not enforce them.
+        if inner.validation_rules(pl.col(alias or "")):
+            raise ValueError(
+                "Validation rules on the inner type of Array are not yet supported."
+            )
+
+        super().__init__(
+            nullable=nullable,
+            primary_key=False,
+            check=check,
+            alias=alias,
+            metadata=metadata,
+        )
+        self.inner = inner
+        self.shape = self._ensure_tuple(shape)
+
+    @property
+    def dtype(self) -> pl.DataType:
+        return pl.Array(self.inner.dtype, self.shape)
+
+    @staticmethod
+    def _ensure_tuple(value: int | tuple[int, ...]) -> tuple[int, ...]:
+        if isinstance(value, tuple):
+            return value
+        return (value,)
+
+    def sqlalchemy_dtype(self, dialect: sa.Dialect) -> sa_TypeEngine:
+        # NOTE: We might want to add support for PostgreSQL's ARRAY type or use JSON in the future.
```

Comment:
👍🏼 I think we're generally still lacking support for more advanced dtypes across various RDBMS 😅